packages/web/src/components/comments/CommentActionBar.tsx

```diff
@@ -0,0 +1,195 @@
+  const isUserGettingNotifs = true // TODO: Need to set up API to provide this
+  const notificationsMuted = false // TODO: Need to set up API to provide this
+
+  const handleCommentReact = useCallback(() => {
+    setReactionState(!reactionState)
+    reactToComment(commentId, !reactionState)
+  }, [commentId, reactToComment, reactionState])
+
+  const handleCommentDelete = useCallback(() => {
+    onClickDelete()
+    setIsDeleting(true)
+    deleteComment(commentId)
+  }, [commentId, deleteComment, onClickDelete])
+
+  const handleCommentPin = useCallback(() => {
+    pinComment(commentId, !isPinned)
+  }, [commentId, isPinned, pinComment])
+
+  const popupMenuItems = useMemo(() => {
+    let items: PopupMenuItem[] = []
+    const entityOwnerMenuItems: PopupMenuItem[] = [
+      {
+        onClick: handleCommentPin,
+        text: messages.pin(isPinned),
+        icon: <IconPin />
+      }
+    ]
+    const commentOwnerMenuItems: PopupMenuItem[] = [
+      { onClick: onClickEdit, text: messages.edit, icon: <IconPencil /> }
+    ]
+    const nonCommentOwnerItems: PopupMenuItem[] = [
+      {
+        onClick: () => {}, // TODO - nothing implemented yet
+        text: messages.report,
+        icon: <IconShieldCheck /> // TODO: temporary icon
+      },
+      { onClick: () => {}, text: messages.block, icon: <IconUserUnfollow /> } // TODO - nothing implemented yet
+    ]
+    const muteNotifs: PopupMenuItem = {
+      onClick: () => {}, // TODO - nothing implemented yet here
+      text: messages.muteNotifs(notificationsMuted),
+      icon: <IconNotificationOff />
+    }
+    const deleteComment: PopupMenuItem = {
+      onClick: handleCommentDelete,
+      text: messages.delete,
+      icon: <IconTrash />
+    }
+
+    if (isEntityOwner) {
+      items = items.concat(entityOwnerMenuItems)
+    }
+    if (isCommentOwner) {
+      items = items.concat(commentOwnerMenuItems)
+      if (isUserGettingNotifs) {
+        items.push(muteNotifs)
+      }
+    }
+    if (!isCommentOwner) {
+      items = items.concat(nonCommentOwnerItems)
+    }
+    if (isCommentOwner || isEntityOwner) {
+      items.push(deleteComment)
+    }
+    return items
+  }, [
+    isCommentOwner,
+    isEntityOwner,
+    isPinned,
+    isUserGettingNotifs,
+    notificationsMuted,
+    onClickEdit,
+    handleCommentDelete,
+    handleCommentPin
+  ])
+
+  return (
+    <Flex gap='l' alignItems='center'>
+      <Flex alignItems='center'>
+        <IconButton
+          icon={IconHeart}
+          color={reactionState ? 'active' : 'subdued'}
+          aria-label='Heart comment'
+          onClick={handleCommentReact}
+          disabled={isDisabled}
+        />
+        <Text color={isDisabled ? 'subdued' : 'default'}> {reactCount}</Text>
+      </Flex>
+      <TextLink
+        variant='subdued'
+        onClick={onClickReply}
+        size='m'
+        disabled={isDisabled}
+      >
+        Reply
+      </TextLink>
+
+      <PopupMenu
```

Comment:
I would assume they want the popup aligned left bottom similar to our track tile overflow menus

Reply:
I'll add it to a list of qa things to have design look at 👀 Planning to do some early desktop design QA soon